storage/aptosdb/src/lib.rs

```diff
@@ -1335,10 +1335,56 @@ impl DbWriter for AptosDB {
 
     fn get_state_snapshot_receiver(
         &self,
-        version: Version,
-        expected_root_hash: HashValue,
+        target_ledger_info: LedgerInfoWithSignatures,
+        target_output_with_proof: TransactionOutputListWithProof,
     ) -> Result<Box<dyn StateSnapshotReceiver<AccountStateBlob>>> {
         gauged_api("get_state_snapshot_receiver", || {
+            // Get the target version and expected root hash
+            let version = target_ledger_info.ledger_info().version();
+            let expected_root_hash = target_output_with_proof
+                .proof
+                .transaction_infos
+                .first()
+                .expect("Target transaction info should exist!")
+                .state_change_hash();
+
+            // Save the target ledger info
+            utils::save_ledger_infos(
+                self.db.clone(),
+                self.ledger_store.clone(),
+                &[target_ledger_info],
+            )?;
```

Comment:
At this point we should really make sure all the epoch ending ledger infos to be available in every node -- assuming we don't publish any waypoints other than the genesis, trust from any downstream can only be established with all of those available.

(or maybe I misread it, and you did that outside of this?)

Reply:
Yes, so this already happens in the bootstrapper. Before we get to syncing any data, the bootstrapper will fetch all epoch changes and verify them, e.g.,: https://github.com/aptos-labs/aptos-core/blob/main/state-sync/state-sync-v2/state-sync-driver/src/bootstrapper.rs#L32. 

This function only gets given the most recent to write to the DB. So, we're holding all epoch changes in memory but only writing the latest when we do the account state sync. oes this make sense?